ulauncher/ui/windows/ulauncher_window.py

```diff
@@ -55,6 +55,20 @@ def __init__(self, **kwargs: Any) -> None:
             **kwargs,
         )
 
+        # Initialize layer shell BEFORE any window operations
+        # This must happen before the window is mapped to avoid Niri issues
+        self.layer_shell_enabled = False
+        if layer_shell.is_supported():
+            try:
+                self.layer_shell_enabled = layer_shell.enable(self)
+                if self.layer_shell_enabled:
+                    logger.info("Layer shell enabled successfully")
+                else:
+                    logger.warning("Layer shell initialization failed, falling back to standard window")
+            except Exception as e:
+                logger.warning(f"Layer shell failed with error: {e}, using standard window")
+                self.layer_shell_enabled = False
+
```

Comment:
Fixed in 41595b91 and 7711fd12 now (without the try/except, but there should be no need for it).